language/move-vm/runtime/src/interpreter.rs

```diff
@@ -235,22 +234,13 @@ impl<'txn> Interpreter<'txn> {
         function: Arc<Function>,
         ty_args: Vec<Type>,
     ) -> VMResult<()> {
-        let module_id = function
-            .module_id()
-            .expect("Module must exist on native function");
-        let function_name = function.name();
-        let native_function =
-            FunctionResolver::resolve(module_id, function_name).ok_or_else(|| {
-                VMStatus::new(StatusCode::LINKER_ERROR)
-                    .with_message(format!("Cannot find native function {}", function_name))
-            })?;
-
         let mut arguments = VecDeque::new();
-        let expected_args = native_function.num_args();
+        let expected_args = function.arg_count();
```

Comment:
no reason to ask the native function, in fact it's risky that if there was a mismatch we would not push or pop correct number of values into/from